### DISCUSSION TITLE
Notifications - Not all applications listed

### DISCUSSION THREAD

**Discussion by aplumez** [andyholmes/gnome-shell-extension-gsconnect#1586] [Q&A] [answered]
Hello!

I installed GSConnect from https://extensions.gnome.org/ and almost everything is working fine.

I would like my Google Chrome notifications on my desktop (Ubuntu 22.04) to be sent to my Android device, the problem is that I don't see Google Chrome in the list of applications in "GSConnect Preferences > Devices > Notifications" pane, so I can't enable it.

Do you have any idea what could be missing?

Thank you!

**Comment by ferdnyc** [comment]
@aplumez 

Chrome doesn't show up in the list for me, either. And, testing with https://www.bennish.net/web-notifications.html, notifications it creates aren't picked up by GSConnect. I **think** I see why.

GSConnect watches for notifications being created, and attempts to connect them back to a source application. The ones it's successful for, it adds to the Applications list for Notifications. It attempts to trace a notification back to its source application by using the "appName" field of the notification.

But when it comes to Chrome notifications (and probably others like them, which use the freedesktop Notifications interface), we're not looking at the correct field of the message, so we're not correctly matching it up to Chrome as the source.

I've opened issue #1591 about the detection bug.

**Comment by ferdnyc** [comment, marked as answer]
@aplumez 

This should now be fixed in the development code. If you'd like to test it out, you can download a build of GSConnect with the fix applied from this page: https://github.com/GSConnect/gnome-shell-extension-gsconnect/actions/runs/4613247847

By clicking on `gsconnect@andyhomes.github.io` in the Artifacts section at the bottom. The extension will download as a .zip file that you can install from a terminal session:
```sh
cd /path/to/downloaded/file
gnome-extensions install --force gsconnect@andyholmes.github.io.zip
```

And then you'll probably need to log out and log back in, to activate the updated extension.

The artifact link is good for some fixed amount of time, I can't remember how long but I think it's pretty long — 90 days, or something like that.

**Reply by aplumez** [reply]
Thank you for your answers @ferdnyc!

I just installed it, but it is showing the extension as "error" in https://extensions.gnome.org/local/. Could it be related to the fact that I have gnome-shell version 43.1 ?

**Reply by ferdnyc** [reply]
@aplumez 

No, that should be fine. Ahh, shoot, I just tried it -- it's an issue with the permissions that we'd never tested. These commands will get things working again. You can just copy-paste them into a terminal session.

```sh
chmod +x ~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/gsconnect-preferences
chmod +x ~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/nautilus-gsconnect.py
chmod +x ~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/daemon.js
chmod +x ~/.local/share/gnome-shell/extensions/gsconnect@andyholmes.github.io/service/nativeMessagingHost.js
```

Sorry about that, I guess we hadn't tested the GitHub-generated artifacts. I'll put in a fix to generate them with the correct permissions in the future.

After you apply the permissions changes, you should be able to open the preferences window, and the Reload button inside should refresh the connection. If it's still not working, a restart should kick it for sure, though I didn't even have to do that on my system. But I'm running 43.3, the current Fedora 37 build, and everything seems to be working fine now that the permissions are fixed.

**Reply by ferdnyc** [reply]
Er, wait, sorry, let me correct those last two.

**Reply by ferdnyc** [reply]
OK, NOW you can copy-paste them into a terminal. :grin:

**Reply by aplumez** [reply]
Yay, working fine! I got my notification forwarded, thank you!

**Reply by ferdnyc** [reply]
Glad to hear it! I nearly have a fix ready for the issues with permissions when installing test extensions from GitHub Actions artifacts, so that hopefully won't bite anyone else in the future.

In the meantime, I'm going to close this discussion as completed. But feel free to open a new one, or file bug reports in Issues, if you run into any more problems with notification-forwarding (or anything else).

**Reply by ferdnyc** [reply]
@aplumez 

So, turns out I was wrong about the GNOME versioning. That build of GSConnect linked above is from after the codebase dropped support for GNOME Shell 43, and even if things are working now, you may hit troubles next time you restart. (The Quick Settings button code uses new APIs, because GNOME 44 deprecated some of the ones from 
GNOME 43, and replaced them with new ones that didn't exist in GNOME 43.)

This build of GSConnect is made from the version 54 code, but has the notifications fix added in. I'd recommend `gnome-extensions install --force`-ing it over the version you currently have installed, just to be safe.

[gsconnect@andyholmes.github.io.zip](https://github.com/GSConnect/gnome-shell-extension-gsconnect/files/11163824/gsconnect%40andyholmes.github.io.zip)
